src/status/condition/operations.rs

```diff
@@ -0,0 +1,150 @@
+use crate::commons::cluster_operation::ClusterOperation;
+use crate::status::condition::{
+    ClusterCondition, ClusterConditionSet, ClusterConditionStatus, ClusterConditionType,
+    ConditionBuilder,
+};
+
+/// Default implementation to build [`ClusterCondition`]s for
+/// the ClusterOperation.
+#[derive(Debug, Clone)]
+pub struct ClusterOperationsConditionBuilder<'a> {
+    cluster_operation: &'a ClusterOperation,
+}
+
+impl<'a> ConditionBuilder for ClusterOperationsConditionBuilder<'a> {
+    fn build_conditions(&self) -> ClusterConditionSet {
+        vec![self.reconciliation_paused(), self.cluster_stopped()].into()
+    }
+}
+
+impl<'a> ClusterOperationsConditionBuilder<'a> {
+    pub fn new(cluster_operation: &'a ClusterOperation) -> Self {
+        Self { cluster_operation }
+    }
+
+    /// Returns the `ReconciliationPaused` cluster condition.
+    fn reconciliation_paused(&self) -> ClusterCondition {
```

Comment:
Happy to leave this as is, just wondering why this in not part of the trait implementation? Since this returns Conditions it seems to be logically tied to the trait anyway and there is no real need to have it in the main impl?

Reply:
We debated this also during the implementation.

The interface (trait) was kept minimal because:
1. Of the implications: clients might query the paused condition but not others? Should other condition types get their own dedicated interfaces ? What about future conditions ? We didn't see the benefit here.
2. Not all resources with a status have a meaningful `ReconciliationPaused` condition. For example: ZNodes, DBConnections. We don't implement the status for those right now, but it's possibly a refactoring task for the future. Airflow and Supreset would benefit from it.

Reply:
ok